onnxruntime/contrib_ops/webgpu/quantization/subgroup_matrix_matmul_nbits.cc

```diff
@@ -73,23 +73,80 @@ bool IsSubgroupMatrixConfigSupportedOnIntel(onnxruntime::webgpu::ComputeContext&
   return false;
 }
 
+// This program optimizes the layout of input matrix A(MxK) for SubgroupMatrixLoad, so that all elements of each
+// subgroup matrix(mxk) are arranged continuously in memory.
+// Take "M = 4, K = 4, m = 2, k = 2" as an example, the input matrix A is arranged in row-major order as follows:
+// d00, d01, | d02, d03,
+// d10, d11, | d12, d13,
+// ---------------------
+// d20, d21, | d22, d23,
+// d30, d31, | d32, d33,
+//
+// The layout program rearranges the input matrix A to be in the following order:
+// d00, d01,
```

Comment:
// d00, d01, d10, d11
-----------------------
// d02, d03, d12, d13

does writing it like this explain it better ?

Reply:
In fact, I like the current data arrangement, which clearly show that output's subgroup matrix's stride becomes 2 from 4. If we use subgroupMatrixLoad, it will load the data d00, d01,d10, d11.

But if we list the output like below:
// d00, d01, d10, d11
// d02, d03, d12, d13
...

I may misunderstand that subgroupMatrixLoad may load the data d00, d01, d02, d03 as a subgroupMatrix.